astro/cli/astro/cmd/cmds.go

```diff
@@ -23,112 +23,137 @@ import (
 
 	"github.com/spf13/cobra"
 	"github.com/uber/astro/astro"
+	"github.com/uber/astro/astro/logger"
 )
 
-var (
-	detach            bool
-	moduleNamesString string
-)
+func (cli *AstroCLI) createRootCommand() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:               "astro",
+		Short:             "A tool for managing multiple Terraform modules.",
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+		PersistentPreRunE: cli.preRun,
+	}
 
-var applyCmd = &cobra.Command{
-	Use:                   "apply [flags] [-- [Terraform argument]...]",
-	DisableFlagsInUseLine: true,
-	Short:                 "Run Terraform apply on all modules",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		c, err := currentProject()
-		if err != nil {
-			return err
-		}
-
-		vars := flagsToUserVariables()
-
-		var moduleNames []string
-		if moduleNamesString != "" {
-			moduleNames = strings.Split(moduleNamesString, ",")
-		}
-
-		status, results, err := c.Apply(
-			astro.ApplyExecutionParameters{
-				ExecutionParameters: astro.ExecutionParameters{
-					ModuleNames:         moduleNames,
-					UserVars:            vars,
-					TerraformParameters: args,
-				},
-			},
-		)
-		if err != nil {
-			return fmt.Errorf("ERROR: %v", processError(err))
-		}
+	rootCmd.PersistentFlags().BoolVarP(&cli.flags.verbose, "verbose", "v", false, "verbose output")
+	rootCmd.PersistentFlags().BoolVarP(&cli.flags.trace, "trace", "", false, "trace output")
+	rootCmd.PersistentFlags().StringVar(&cli.flags.userCfgFile, "config", "", "config file")
+
+	return rootCmd
+}
+
+func (cli *AstroCLI) createApplyCmd() *cobra.Command {
```

Comment:
For all these `createXXXCmd`, since they are methods (with a receiver object), they should not have return values. Instead, they should just initialize the receiver object, like 
```
# we can return errors if necessary
func (cli *AstroCLI) createApplyCmd() {
  applyCmd := ...
  ...

  cli.apply = applyCmd
}
```

Currently we are using constructor function coding style to implement initialization method.

Reply:
👍will fix

Reply:
Fixed.